### ISSUE TITLE
Style/MutableConstant advises "freezing" instance of Class

### ISSUE THREAD

**Issue by rhys-vdw** [rubocop-hq/rubocop#6000]
I have no idea what `Class#freeze` does but it seems unusual.

--------

## Expected behavior

No error raised

## Actual behavior

Error raised:

```
path/to/file:2 W: Style/MutableConstant: Freeze mutable objects assigned to constants.
```

## Steps to reproduce the problem

```rb
class X
  Y = Struct.new(:z)
end
```

## RuboCop version

```
-> bundle exec rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-darwin16)
```

**Comment by rhys-vdw** [comment]
More suprising: The implied code still triggers the same rule violation:

```
class X
  Y = Struct.new(:z).freeze
end
```

**Comment by rhys-vdw** [comment]
Oops, this was wrong. Misread file path in error report. :-S
